Source/WebCore/platform/mac/ScrollbarThemeMac.mm

```diff
@@ -438,9 +438,12 @@ static IntRect buttonRepaintRect(const IntRect& buttonRect, ScrollbarOrientation
 
 int ScrollbarThemeMac::minimumThumbLength(Scrollbar& scrollbar)
 {
-    BEGIN_BLOCK_OBJC_EXCEPTIONS
-    return [scrollbarMap().get(&scrollbar) knobMinLength];
-    END_BLOCK_OBJC_EXCEPTIONS
+    if (scrollbar.shouldRegisterScrollbar()) {
```

Comment:
This stuff makes me wonder why we don't just ask the scrollbar for its minimumKnobLength. Why is the map lookup needed? Then we could have a polymorphic Scrollbar class where there's a subclass that never has an NSScrollerImp.

Reply:
Do you mean rather than having the map, have the Scrollbar class hold the NSScrollerImp if UI-side compositing is off, and have a Scrollbar not held if UI-side compositing is on? If yes then yeah I think that would be a good cleanup for this area once this lands.